### PR TITLE
update htsjdk to 2.19.0

### PR DESCRIPTION
**Pull request by lbergelson**
Changed necessary for htsjdk 2.19.0, but we're blocked by https://github.com/disq-bio/disq/issues/95.

We might also want to wait for https://github.com/broadinstitute/picard/pull/1297 to be incorporated although I think it's ok not to wait.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -54,16 +54,19 @@ repositories {
         url "https://broadinstitute.jfrog.io/broadinstitute/libs-snapshot/" //for htsjdk snapshots
     }
 
+    maven {
```
Comment (lbergelson) [inline build.gradle]: This can go once we update to the new disq.

File: src/main/java/org/broadinstitute/hellbender/utils/io/IOUtils.java [hunk 2]
```diff
@@ -1021,13 +1022,7 @@ public static void deleteOnExit(final Path fileToDelete){
...
-    public static void deleteRecursively(final Path rootPath) throws IOException {
-        final List<Path> pathsToDelete = Files.walk(rootPath)
-                        .sorted(Comparator.reverseOrder())
-                        .collect(Collectors.toList());
-
-        for (Path path : pathsToDelete) {
-            Files.deleteIfExists(path);
-        }
+    public static void deleteRecursively(final Path rootPath) {
+        IOUtil.recursiveDelete(rootPath);
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/utils/io/IOUtils.java]: New implementation in htsjdk.  We might want to just remove this one actually.

File: src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java [hunk 3]
```diff
@@ -209,7 +209,8 @@ public void testBCFIndex() {
         checkIndex(index, Arrays.asList("1"));
     }
 
-    @Test(expectedExceptions = UserException.CouldNotIndexFile.class)
+    // test disabled until https://github.com/samtools/htsjdk/issues/1323 is resolved
+    @Test(enabled = false)
```
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java]: this is resolved but needs yet another htsjdk release to fix.

**Comment by lbergelson** [comment]
This should add support for reading bam files with CSI indexes, as well as porting the FastaReferenceWriter to htsjdk and a lot of other changes to htsjdk.  @samuelklee This includes the overlap detector optimizations you wanted as well as the changes to let you write interval file to paths.

**Comment by tomwhite** [comment]
Disq 0.3.0 has now been released.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/fasta/FastaReferenceMaker.java [hunk 4]
```diff
@@ -82,7 +83,10 @@
     public void onTraversalStart() {
         final Path path = IOUtils.getPath(output);
         try {
-            writer = new FastaReferenceWriter(path, basesPerLine, true, true);
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/fasta/FastaReferenceMaker.java]: Is the behavior previously controlled by the two `true` booleans here being preserved in the new `FastaReferenceWriterBuilder` call?

File: src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java [hunk 5]
```diff
@@ -209,7 +209,8 @@ public void testBCFIndex() {
         checkIndex(index, Arrays.asList("1"));
     }
 
-    @Test(expectedExceptions = UserException.CouldNotIndexFile.class)
+    // test disabled until https://github.com/samtools/htsjdk/issues/1323 is resolved
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java]: Do we have a GATK issue to track this, and remind us to re-enable this test in the future?

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/fasta/FastaReferenceMaker.java (hunk 4 above)
Comment (cmnbroad) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/fasta/FastaReferenceMaker.java]: Looks like these were `makeFai` and `makeDict`, both of which are `true` by default in the new builder.

File: src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java (hunk 5 above)
Comment (cmnbroad) [reply src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java]: https://github.com/broadinstitute/gatk/issues/5838

**Comment by cmnbroad** [comment]
This also includes an upgrade to Picard 2.19, so its in sync with htsjdk now.

**Pull request merged**